trustpinning/certs_test.go

```diff
@@ -1007,12 +1003,70 @@ func generateExpiredTestingCertificate(rootKey data.PrivateKey, gun string) (*x5
 	return cryptoservice.GenerateCertificate(rootKey, gun, startTime, startTime.AddDate(1, 0, 0))
 }
 
+func TestParsePEMPublicKey(t *testing.T) {
+	gun := "notary"
+	memStore := trustmanager.NewKeyMemoryStore(passphraseRetriever)
+	cs := cryptoservice.NewCryptoService(memStore)
+
+	// can parse ECDSA PEM
+	ecdsaPubKey, err := cs.Create("root", "docker.io/notary/test", data.ECDSAKey)
+	require.NoError(t, err)
+	ecdsaPemBytes := pem.EncodeToMemory(&pem.Block{
+		Type:    "PUBLIC KEY",
+		Headers: nil,
+		Bytes:   ecdsaPubKey.Public(),
+	})
+
+	ecdsaParsedPubKey, err := utils.ParsePEMPublicKey(ecdsaPemBytes)
+	require.NoError(t, err, "no key: %s", ecdsaParsedPubKey.Public())
+
+	// can parse certificates
+	ecdsaPrivKey, _, err := memStore.GetKey(ecdsaPubKey.ID())
+	require.NoError(t, err)
+	cert, err := generateTestingCertificate(ecdsaPrivKey, gun, notary.Day*30)
+	require.NoError(t, err)
+	ecdsaPubKeyFromCert, err := utils.ParsePEMPublicKey(utils.CertToPEM(cert))
+	require.NoError(t, err)
+
+	thatData := []byte{1, 2, 3, 4}
+	sig, err := ecdsaPrivKey.Sign(rand.Reader, thatData, nil)
+	require.NoError(t, err)
+	err = signed.ECDSAVerifier{}.Verify(ecdsaPubKeyFromCert, sig, thatData)
+	require.NoError(t, err)
+
+	// can parse RSA PEM
+	rsaPubKey, err := cs.Create("root", "docker.io/notary/test2", data.RSAKey)
+	require.NoError(t, err)
+	rsaPemBytes := pem.EncodeToMemory(&pem.Block{
+		Type:    "PUBLIC KEY",
+		Headers: nil,
+		Bytes:   rsaPubKey.Public(),
+	})
+	_, err = utils.ParsePEMPublicKey(rsaPemBytes)
+	require.NoError(t, err)
+
+	// unsupported key type
+	unsupportedPemBytes := pem.EncodeToMemory(&pem.Block{
+		Type:    "UNSUPORTED KEY",
```

Comment:
I think this case errors because `pem.Decode` fails, not because of the type-switch.

This case is great, but It would be awesome if we could add another test case with a valid type that `ParsePEMPublicKey` does not support, like `CERTIFICATE REQUEST` or `PRIVATE KEY`